File: Leetcode/1200.cpp
class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
        sort(arr.begin(), arr.end());
        int minVal = numeric_limits<int>::max();
        int n = (int)arr.size();
        for (int i = 0; i < n - 1; ++i) {
            minVal = min(minVal, arr.at(i + 1) - arr.at(i));
        }
        vector<vector<int> > result;
        for (int i = 0; i < n - 1; ++i) {
            if (arr.at(i + 1) - arr.at(i) == minVal) {
                result.emplace_back(vector<int>{arr.at(i),arr.at(i+1)});
            }
        }
        return result;
    }
};

File: Leetcode/1200/1200.go
func min(a, b int) int {
	if a > b {
		return b
	}
	return a
}

func minimumAbsDifference(arr []int) [][]int {
	sort.Ints(arr)
	minDiff := math.MaxInt32
	for i := 1; i < len(arr); i++ {
		minDiff = min(minDiff, arr[i]-arr[i-1])
	}
	unmap := make(map[int]struct{})
	value := struct {
	}{}
	for i := 0; i < len(arr); i++ {
		unmap[arr[i]] = value
	}
	var ans [][]int
	for i := 0; i < len(arr); i++ {
		_, ok := unmap[arr[i]+minDiff]
		if ok {
			ans = append(ans, []int{arr[i], arr[i]+minDiff})
		}
	}
	return ans
}